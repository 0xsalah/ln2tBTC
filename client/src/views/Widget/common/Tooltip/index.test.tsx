import React from "react";
import { render } from "@testing-library/react";
import { Tooltip, Hint } from ".";

test("snapshot of component rendered hint question mark (default)", () => {
  const component = render(<Hint />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered hint custom logo", () => {
  const component = render(<Hint svgIcon={
    <svg>
      <symbol id="icon-clock" viewBox="0 0 47 47">
        <circle cx="23.5" cy="23.5" r="22" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
        <path d="M23.5 9.5V23.5H37.5" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
      </symbol>
    </svg>
  } />);
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of tooltip", () => {
  const myRef = React.createRef<HTMLSpanElement>();
  const component = render(
    <Tooltip hintButton={myRef} onDismiss={() => null} >
      Hint
    </Tooltip>
  );
  expect(component.baseElement).toMatchSnapshot();
});