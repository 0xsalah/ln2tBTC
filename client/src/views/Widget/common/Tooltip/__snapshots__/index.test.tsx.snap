// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered hint custom logo 1`] = `
<body>
  <div>
    <span
      class="tooltip-general"
    >
      <svg>
        <symbol
          id="icon-clock"
          viewBox="0 0 47 47"
        >
          <circle
            cx="23.5"
            cy="23.5"
            r="22"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M23.5 9.5V23.5H37.5"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </symbol>
      </svg>
    </span>
  </div>
</body>
`;

exports[`snapshot of component rendered hint question mark (default) 1`] = `
<body>
  <div>
    <span
      class="tooltip-help"
    >
      <svg
        class="icon icon-question"
      >
        <use
          xlink:href="icons.svg#icon-question"
        />
      </svg>
    </span>
  </div>
</body>
`;

exports[`snapshot of tooltip active 1`] = `
<body>
  <div>
    <div
      class="tooltipster-base tooltipster-sidetip tooltipster-0 fff tooltipster-fade tooltipster-show"
      style="position: absolute; max-width: 340px; max-height: 400px; z-index: 99; top: -1000px; left: -1000px; animation-duration: 350ms; transition-duration: 350ms;"
    >
      <div
        class="tooltipster-box"
      >
        <div
          class="tooltipster-content"
        >
          Hint
        </div>
      </div>
      <div
        class="tooltipster-arrow"
        style="left: 0px;"
      >
        <div
          class="tooltipster-arrow-uncropped"
        >
          <div
            class="tooltipster-arrow-border"
          />
          <div
            class="tooltipster-arrow-background"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot of tooltip hidden 1`] = `
<body>
  <div>
    <div
      class="tooltipster-base tooltipster-sidetip tooltipster-0 fff tooltipster-fade tooltipster-dying"
      style="position: absolute; max-width: 0px; max-height: 0px; z-index: 99; top: 0px; left: 0px; animation-duration: 350ms; transition-duration: 350ms;"
    >
      <div
        class="tooltipster-box"
      >
        <div
          class="tooltipster-content"
        >
          Hint
        </div>
      </div>
      <div
        class="tooltipster-arrow"
        style="left: 0px;"
      >
        <div
          class="tooltipster-arrow-uncropped"
        >
          <div
            class="tooltipster-arrow-border"
          />
          <div
            class="tooltipster-arrow-background"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
